mqtt/mqtt3/src/client/mod.rs

```diff
@@ -337,6 +337,12 @@ where
                                 }
 
                                 connect.reconnect();
+
+                                if err.is_connection_error() {
+                                    return std::task::Poll::Ready(Some(Ok(Event::Disconnected(
+                                        format!("Connection failed {}", err),
```

Comment:
Why not return the `EncodeError` itself instead of a `String` ?

Reply:
added ConnectionError